index.js

```diff
@@ -18,6 +14,53 @@ class VPCPlugin {
   }
 
   /**
+   * Gets the desired vpc with the designated subnets and security groups
+   * that were set in serverless config file
+   * @returns {Promise}
+   */
+  updateVpcConfig() {
+    const awsCreds = this.serverless.providers.aws.getCredentials();
+
+    AWS.config.update(awsCreds);
+    this.ec2 = new AWS.EC2();
```

Comment:
It looks like `AWS.EC2` takes a `credentials` argument. Can you try passing it in that way instead of using `AWS.config.update`?

http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/EC2.html